### PR TITLE
[X86][test] Remove useless pattern for VDPBF16PSZmb and add a test for broadcast folding

### PR DESCRIPTION
**Pull request by KanRobert**
llvm-issue: https://github.com/llvm/llvm-project/issues/68810

### CODE REVIEW

**Comment by bjacob** [comment]
I'll let the maintainers do the actual review, just want to say thanks and yes, the added test does correspond to what https://github.com/llvm/llvm-project/issues/68810 is about. Thanks!

**Review by phoebewang — APPROVED** [review]
LGTM, thanks!

**Pull request merged**
